### PR TITLE
[WIP] Embedded cli

### PR DESCRIPTION
**Pull request by oriooctopus**
Please do not review this. It is a work in progress

### CODE REVIEW

**Comment by libra-action** [comment]
**LSR dependency change summary**:
```
target packages:
  A cpuid-bool 0.2.0 (transitive third-party, crates.io) (other versions: 0.1.2)
    * features: [none]
  M bcs 0.1.2 (direct third-party, crates.io)
    * version upgraded from 0.1.1
    * (unchanged features: [none])
  M openssl-sys 0.9.59 (transitive third-party, crates.io)
    * version upgraded from 0.9.58
    * (unchanged features: [none])
  M parking_lot 0.11.1 (transitive third-party, crates.io)
    * version upgraded from 0.11.0
    * (unchanged features: default)
  M polyval 0.4.3 (transitive third-party, crates.io)
    * version upgraded from 0.4.1
    * (unchanged features: [none])
```

**Comment by libra-action** [comment]
**release binaries dependency change summary**:
```
target packages:
  A cpuid-bool 0.2.0 (transitive third-party, crates.io) (other versions: 0.1.2)
    * features: [none]
  M bcs 0.1.2 (direct third-party, crates.io)
    * version upgraded from 0.1.1
    * (unchanged features: [none])
  M crossbeam-epoch 0.9.1 (transitive third-party, crates.io)
    * version upgraded from 0.9.0
    * (unchanged features: alloc, lazy_static, std)
  M libc 0.2.81 (transitive third-party, crates.io)
    * added features: align
    * (unchanged features: default, std)
  M memoffset 0.6.1 (transitive third-party, crates.io)
    * version upgraded from 0.5.6
    * (unchanged features: default)
  M openssl-sys 0.9.59 (transitive third-party, crates.io)
    * version upgraded from 0.9.58
    * (unchanged features: [none])
  M parking_lot 0.11.1 (transitive third-party, crates.io)
    * version upgraded from 0.11.0
    * (unchanged features: default)
  M polyval 0.4.3 (transitive third-party, crates.io)
    * version upgraded from 0.4.1
    * (unchanged features: [none])
  M socket2 0.3.18 (transitive third-party, crates.io)
    * version upgraded from 0.3.15
    * (unchanged features: [none])
```

**Comment by libra-action** [comment]
**LEC dependency change summary**:
```
target packages:
  A cpuid-bool 0.2.0 (transitive third-party, crates.io) (other versions: 0.1.2)
    * features: [none]
  M bcs 0.1.2 (direct third-party, crates.io)
    * version upgraded from 0.1.1
    * (unchanged features: [none])
  M crossbeam-epoch 0.9.1 (transitive third-party, crates.io)
    * version upgraded from 0.9.0
    * (unchanged features: alloc, lazy_static, std)
  M libc 0.2.81 (transitive third-party, crates.io)
    * added features: align
    * (unchanged features: default, std)
  M memoffset 0.6.1 (transitive third-party, crates.io)
    * version upgraded from 0.5.6
    * (unchanged features: default)
  M openssl-sys 0.9.59 (transitive third-party, crates.io)
    * version upgraded from 0.9.58
    * (unchanged features: [none])
  M parking_lot 0.11.1 (transitive third-party, crates.io)
    * version upgraded from 0.11.0
    * (unchanged features: default)
  M polyval 0.4.3 (transitive third-party, crates.io)
    * version upgraded from 0.4.1
    * (unchanged features: [none])
  M socket2 0.3.18 (transitive third-party, crates.io)
    * version upgraded from 0.3.15
    * (unchanged features: [none])
```

**Comment by davidiw** [comment]
fwiw, we don't really pull in merge commits and instead have a linear history

**Comment by JoelMarcey** [comment]
Are we still keeping this PR, @oriooctopus ?

File: developers.diem.com/src/components/docs/CLI/index.js [hunk 1]
```diff
@@ -1,12 +1,24 @@
...
-import './styles.css';
-import classnames from 'classnames';
-import styles from './styles.module.css';
+import utils from 'diem-docusaurus-components/src/utils';
+
+import commandsStyles from './css/commands.module.css';
+import generalStyles from './css/general.module.css';
+import tutorialStyles from './css/tutorial.module.css';
+
+const {mergeCSSModules} = utils;
```
Comment (jjwill) [inline developers.diem.com/src/components/docs/CLI/index.js:14]: nit: Space after `{` and before `}` (lets manage this in linter)

File: developers.diem.com/src/css/custom.css [hunk 2]
```diff
@@ -8,7 +8,11 @@ iframe {
   border: 0;
 }
 
-/* Doc Classes */
+.main-content {
```
Comment (jjwill) [inline developers.diem.com/src/css/custom.css:11]: It looks like we should be using the grid here but we don't need to change this right now.

**Review by jjwill — APPROVED** [review]
Couple comments but LGTM

**Comment by davidiw** [comment]
this PR has been open for nearly 2 months without any activity, closing

**Pull request closed without merging**